filament/backend/src/vulkan/VulkanCommands.cpp

```diff
@@ -271,6 +282,93 @@ void VulkanCommands::updateFences() {
     }
 }
 
+void VulkanCommands::beginDebugUtilsLabel(char const* string) {
+    const VkCommandBuffer cmdbuffer = get().cmdbuffer;
+    VkDebugUtilsLabelEXT labelInfo = {
+	.sType = VK_STRUCTURE_TYPE_DEBUG_UTILS_LABEL_EXT,
+	.pLabelName = string,
+	.color = {0, 1, 0, 1},
+    };
+    vkCmdBeginDebugUtilsLabelEXT(cmdbuffer, &labelInfo);
+    mCurrentDebugLabel = string;
+}
+
+void VulkanCommands::insertDebugUtilsLabel(char const* string) {
+    assert_invariant(mCurrent);
+    assert_invariant(mCurrent->cmdbuffer);
+
+    const VkCommandBuffer cmdbuffer = mCurrent->cmdbuffer;
+
+    VkDebugUtilsLabelEXT labelInfo = {
+	.sType = VK_STRUCTURE_TYPE_DEBUG_UTILS_LABEL_EXT,
+	.pLabelName = string,
+	.color = {1, 1, 0, 1},
+    };
+    vkCmdInsertDebugUtilsLabelEXT(cmdbuffer, &labelInfo);
+}
+
+void VulkanCommands::endDebugUtilsLabel() {
+    // TODO: the following should be turned into asserts after refactoring the push/pop
+    // label flow.
+    if (!mCurrent) {
+	return;
+    }
+    if (!mCurrent->cmdbuffer) {
+	return;
+    }
+    if (!mCurrentDebugLabel.empty()) {
```

Comment:
Should this be `if (mCurrentDebugLabel.empty())` ? Same question inside `endDebugMarker`.